### PR TITLE
Demographics & gender

### PR DESCRIPTION
**Pull request by danni**
N.B. This breaks the styling of the radio icons. Fixing this now in material-ui.

### CODE REVIEW

**Comment by danni** [comment]
P.R. to fix customising of RadioButtons https://github.com/callemall/material-ui/pull/1639

File: src/components/TransportTime.js [hunk 1]
```diff
@@ -28,15 +28,29 @@ class TransportTime extends React.Component {
...
+                    <span className="location">
+                        {titleize(this.props.object.location.suburb)}
+                    </span>
+                </div>
+            );
+        } else {
+            /* This is a confidential location, we can't show any
+             * transport time*/
+            return (
+                <div className="TransportTime">
```
Comment (DanielHeath) [inline src/components/TransportTime.js:45]: I'd generally prefer to split this kind of conditional into separate components.

File: src/pages/personalisation/BaseQuestion.js [hunk 2]
```diff
@@ -52,13 +60,25 @@ class BaseQuestion extends React.Component {
...
 
-    onAnswerTouchTap(answer: string, ...rest: any): void {
-        storage.setItem(this.props.name, answer);
+    /**
+     * triggerNext:
+     *
+     * Trigger next page after a 500ms debounce.
+     */
+    /*::__(){`*/@debounce(500)/*::`}*/
+    triggerNext(): void {
```
Comment (DanielHeath) [inline src/pages/personalisation/BaseQuestion.js:69]: Could we use css transitions to give this visual feedback?

File: src/components/TransportTime.js (hunk 1 above)
Comment (danni) [inline src/components/TransportTime.js:45]: Well it's more the component can't do anything. Unless you wanted the component to return ''?

Comment (DanielHeath) [inline src/components/TransportTime.js:45]: I'd probably have it return `<ConfidentialLocation/>` or `<TransportTimeToPoint/>`

Comment (danni) [inline src/components/TransportTime.js:45]: That is too much abstraction IMO. It's simply a case of "can't do what I'm asked, falling back"

File: src/pages/personalisation/BaseQuestion.js (hunk 2 above)
Comment (danni) [inline src/pages/personalisation/BaseQuestion.js:69]: We are. However we also need to have the time to see that.

File: src/components/TransportTime.js (hunk 1 above)
Comment (DanielHeath) [inline src/components/TransportTime.js:45]: Fair enough; maybe if the component were getting huge.

**Comment by DanielHeath** [comment]
++

**Pull request merged**